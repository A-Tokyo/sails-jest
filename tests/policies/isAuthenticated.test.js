describe('isAuthenticated Policy', () => {
  it('should validate the token', () => {
    const req = { param: () => CipherService.sign('user.id.1') };
    const res = {};
    const next = jest.fn();
    sails.middleware.policies.isauthenticated(req, res, next);
    expect(next.mock.calls.length).toBe(1);
    expect(next.mock.calls[0][0]).toBeUndefined();
    expect(req.user).toMatchSnapshot();
  });

  it('should refuse the request', () => {
    const req = { param: () => 'invalid token' };
    const res = { forbidden: jest.fn() };
    const next = jest.fn();
    sails.middleware.policies.isauthenticated(req, res, next);
    expect(next.mock.calls.length).toBe(0);
    expect(res.forbidden.mock.calls.length).toBe(1);
    expect(res.forbidden.mock.calls[0][0]).toMatchSnapshot();
  });
});
