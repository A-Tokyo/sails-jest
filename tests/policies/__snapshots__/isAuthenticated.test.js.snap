// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isAuthenticated Policy should refuse the request 1`] = `[JsonWebTokenError: jwt malformed]`;

exports[`isAuthenticated Policy should validate the token 1`] = `"user.id.1"`;
